tests/unit/metadata/test_single_table.py

```diff
@@ -34,6 +35,544 @@ def test___init__(self):
+
+        Setup:
+            - instance of ``SingleTableMetadata``
+
+        Input:
+            - Column name.
+            - sdtype text
+            - Valid ``regex_format``.
+            - Invalid ``regex_format``.
+
+        Side Effects:
+            - ``ValueError``
+        """
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run / Assert
+        instance._validate_text('phrase', regex_format='[A-z]')
+        error_msg = re.escape("Invalid regex format string '[A-z{' for text column 'phrase'.")
+        with pytest.raises(ValueError, match=error_msg):
+            instance._validate_text('phrase', regex_format='[A-z{')
+
+    def test__validate_column_exists(self):
+        """Test the ``_validate_column_exists`` method.
+
+        Setup:
+            - instance of ``SingleTableMetadata``
+            - A list of ``_columns``.
+
+        Input:
+            - Column name.
+
+        Side Effects:
+            - ``ValueError`` when the column is not in the ``instance._columns``.
+        """
+        # Setup
+        instance = SingleTableMetadata()
+        instance._columns = {
+            'name': {'sdtype': 'categorical'},
+            'age': {'sdtype': 'numerical'},
+            'start_date': {'sdtype': 'datetime'},
+            'phrase': {'sdtype': 'text'},
+        }
+
+        # Run / Assert
+        instance._validate_column_exists('age')
+        error_msg = re.escape(
+            "Column name ('synthetic') does not exist in the table. "
+            "Use 'add_column' to add new column."
+        )
+        with pytest.raises(ValueError, match=error_msg):
+            instance._validate_column_exists('synthetic')
+
+    def test__validate_unexpected_kwargs(self):
+        """Test the ``_validate_unexpected_kwargs`` method.
+
+        Setup:
+            - instance of ``SingleTableMetadata``
+
+        Input:
+            - Column name.
+            - sdtype
+
+        Side Effects:
+            ValueError
+        """
+        # Setup
+        instance = SingleTableMetadata()
+
+        # Run / Assert non raising errors
+        instance._validate_unexpected_kwargs('age', 'numerical')
+        instance._validate_unexpected_kwargs('age', 'numerical', representation='int')
+        instance._validate_unexpected_kwargs('start_date', 'datetime')
+        instance._validate_unexpected_kwargs('start_date', 'datetime', datetime_format='%Y-%d')
+        instance._validate_unexpected_kwargs('name', 'categorical')
+        instance._validate_unexpected_kwargs('name', 'categorical', order_by='alphabetical')
+        instance._validate_unexpected_kwargs('name', 'categorical', order=['a', 'b', 'c'])
+        instance._validate_unexpected_kwargs('synthetic', 'boolean')
+        instance._validate_unexpected_kwargs('phrase', 'text')
+        instance._validate_unexpected_kwargs('phrase', 'text', regex_format='[A-z]')
+        instance._validate_unexpected_kwargs('phone', 'phone_number')
```

Comment:
Can we [parameterize ](https://docs.pytest.org/en/7.1.x/example/parametrize.html)this test instead of calling all these different cases like this? We can make the passed kwargs, and the error message parameters

Reply:
🤔 yes let me re-do it